nmagent/client_test.go

```diff
@@ -622,3 +622,71 @@ func TestGetNCVersionList(t *testing.T) {
 		})
 	}
 }
+
+func TestGetHomeAz(t *testing.T) {
+	tests := []struct {
+		name      string
+		exp       nmagent.HomeAzResponse
+		expPath   string
+		resp      map[string]interface{}
+		shouldErr bool
+	}{
+		{
+			"happy path",
+			nmagent.HomeAzResponse{HomeAz: uint(1)},
+			"/machine/plugins/?comp=nmagent&type=GetHomeAz",
+			map[string]interface{}{
+				"httpStatusCode": "200",
+				"HomeAz":         1,
+			},
+			false,
+		},
+		{
+			"empty response",
+			nmagent.HomeAzResponse{},
+			"/machine/plugins/?comp=nmagent&type=GetHomeAz",
+			map[string]interface{}{
+				"httpStatusCode": "500",
+			},
+			true,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			var gotPath string
+			client := nmagent.NewTestClient(&TestTripper{
+				RoundTripF: func(req *http.Request) (*http.Response, error) {
+					gotPath = req.URL.Path
+					rr := httptest.NewRecorder()
+					err := json.NewEncoder(rr).Encode(test.resp)
+					if err != nil {
+						t.Fatal("unexpected error encoding response: err:", err)
+					}
+					rr.WriteHeader(http.StatusOK)
+					return rr.Result(), nil
+				},
+			})
+
+			got, err := client.GetHomeAz(context.TODO())
+			if err != nil && !test.shouldErr {
```

Comment:
should you be testing if the correct values are returned for IsSupported and AZID?